C/common/asset_tracking.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * FogLAMP asset tracking related
+ *
+ * Copyright (c) 2019 Dianomic, LLC
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Amandeep Singh Arora
+ */
+
+#include <logger.h>
+#include <asset_tracking.h>
+
+using namespace std;
+
+
+AssetTracker *AssetTracker::instance = 0;
+
+AssetTracker *AssetTracker::getAssetTracker()
+{
+	return instance;
+}
+
+AssetTracker::AssetTracker(ManagementClient *mgtClient, string service) 
+	: m_mgtClient(mgtClient), m_service(service)
+{
+	instance = this;
+}
+
+/**
+ * Fetch all asset tracking tuples from DB and populate local cache
+ *
+ * @param service  	Service name
+ * @param plugin  	Plugin name
+ * @param event  	Event name
+ */
+void AssetTracker::populateAssetTrackingCache(string /*plugin*/, string /*event*/)
+{
+	try {
+		std::vector<AssetTrackingTuple*>& vec = m_mgtClient->getAssetTrackingTuples(m_service);
+		for (AssetTrackingTuple* & rec : vec)
+		{
+			/*if (rec->m_pluginName != plugin || rec->m_eventName != event)
```

Comment:
Why is this commented out? If it is not needed then please remove the code.

Reply:
Removed the code for now.